torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2199,6 +2199,28 @@ def sample_inputs_masked_select(op_info, device, dtype, requires_grad):
                                 dtypes=[torch.bfloat16])),
                    safe_casts_outputs=True,
                    handles_complex_extremals=False),
+    UnaryUfuncInfo('square',
+                   ref=np.square,
+                   dtypes=all_types_and_complex_and(torch.bool),
+                   dtypesIfCPU=all_types_and_complex_and(torch.bool),
+                   dtypesIfCUDA=all_types_and_complex_and(torch.bool, torch.half, torch.bfloat16),
+                   skips=(
+                       # Reference: https://github.com/pytorch/pytorch/issues/52549
+                       SkipInfo('TestUnaryUfuncs', 'test_reference_numerics',
+                                dtypes=[torch.cfloat, torch.cdouble]),
+                       # Reference: https://github.com/pytorch/pytorch/pull/52551#issuecomment-782596181
+                       SkipInfo('TestUnaryUfuncs', 'test_reference_numerics',
+                                device_type='cuda', dtypes=[torch.bfloat16]),
+                       # Following error in Windows CI
+                       # File "test_unary_ufuncs.py", line 289, in test_reference_numerics
+                       #  if not torch.can_cast(numpy_to_torch_dtype_dict[expected.dtype.type], dtype):
+                       # KeyError: <class 'numpy.intc'>
```

Comment:
Interesting. Should we be mapping intc to torch.int?

Reply:
`intc` is platform defined.

So we can update the following code like,

https://github.com/pytorch/pytorch/blob/d90d7245f4308f22909f56770719eb4906f54ae2/torch/testing/_internal/common_utils.py#L384-L403

To be something like,
```python
import numpy as np

    # Dict of NumPy dtype -> torch dtype (when the correspondence exists)
    numpy_to_torch_dtype_dict = {
        np.bool_      : torch.bool,
        np.uint8      : torch.uint8,
        np.int8       : torch.int8,
        np.int16      : torch.int16,
        np.int32      : torch.int32,
        np.int64      : torch.int64,
        np.float16    : torch.float16,
        np.float32    : torch.float32,
        np.float64    : torch.float64,
        np.complex64  : torch.complex64,
        np.complex128 : torch.complex128
    }

    if IS_WINDOWS:
        # `int` on Windows is 32bit.
        # https://docs.microsoft.com/en-us/cpp/cpp/data-type-ranges?view=msvc-160#:~:text=The%20int%20and%20unsigned%20int,this%20to%20be%20implementation%2Dspecific.
        numpy_to_torch_dtype_dict[np.intc] = torch.int

    # Dict of torch dtype -> NumPy dtype
    torch_to_numpy_dtype_dict = {value : key for (key, value) in numpy_to_torch_dtype_dict.items()}
```

Also I think, `if TEST_NUMPY` is unnecessary now?

Reply:
You're correct, TEST_NUMPY is uneccessary. The update you're proposing LGTM!

Reply:
Have updated https://github.com/pytorch/pytorch/pull/53181 with the above code.

Reply:
Is removing TEST_NUMPY safe? I think there is an option to build without NumPy right?

Reply:
The test suite has a dependency on NumPy now, so you can't run the tests without NumPy.